src/reanimated2/component/FlatList.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { FlatList, FlatListProps, LayoutChangeEvent } from 'react-native';
+import WrappedComponents from './WrappedComponents';
+import createAnimatedComponent from '../../createAnimatedComponent';
+import { ILayoutAnimationBuilder } from '../layoutReanimation/animationBuilder/commonTypes';
+
+const AnimatedFlatList = createAnimatedComponent(FlatList as any);
+
+const createCellRenderer = (itemLayoutAnimation?: ILayoutAnimationBuilder) => {
+  const cellRenderer: React.FC<{
+    onLayout: (event: LayoutChangeEvent) => void;
+  }> = (props) => {
+    return (
+      <WrappedComponents.View
+        layout={itemLayoutAnimation}
+        onLayout={props.onLayout}>
+        {props.children}
+      </WrappedComponents.View>
+    );
+  };
+
+  return cellRenderer;
+};
+
+interface ReanimatedFlatlistProps<ItemT> extends FlatListProps<ItemT> {
+  itemLayoutAnimation?: ILayoutAnimationBuilder;
+}
+
+type ReanimatedFlatListFC<T = any> = React.FC<ReanimatedFlatlistProps<T>>;
+
+const ReanimatedFlatlist: ReanimatedFlatListFC = ({
+  itemLayoutAnimation,
+  ...restProps
+}) => {
+  const cellRenderer = React.useMemo(
+    () => createCellRenderer(itemLayoutAnimation),
+    []
+  );
+  return (
+    <AnimatedFlatList
+      {...(restProps as any)}
```

Comment:
Why do the types don't work correctly here?

Reply:
Because `FlatList` is generic. @jakub-gonet and I already had a hard time trying to get this working, I'd be surprised if there was a simple solution 😅

Reply:
We had to retype FlatList basically. Checkout https://github.com/software-mansion/react-native-gesture-handler/blob/6cc5e4b8f4523d9e22b5d7aef1b15d3c9dad9cf4/src/components/GestureComponents.tsx#L75-L90 and https://github.com/software-mansion/react-native-gesture-handler/commit/9c7c9115f5baaa8025f1241c068a80f7884a648d.